src/client/Microsoft.Identity.Client/Internal/RequestContext.cs

```diff
@@ -26,12 +26,15 @@ internal class RequestContext
 
         public CancellationToken UserCancellationToken { get; }
 
+        // Add the UseMtlsPop property
+        public bool UseMtlsPop { get; set; }
```

Comment:
Any specific reason to add this in requestContext instead of ApplicationConfiguration?

Reply:
because the WithMtlsProofofPossession() is configured at the request level we are trying to keep the configurations at the request level